airbyte-commons-server/src/test/java/io/airbyte/commons/server/handlers/ConnectorBuilderProjectsHandlerTest.java

```diff
@@ -0,0 +1,150 @@
+
+class ConnectorBuilderProjectsHandlerTest {
+
+  private ConfigRepository configRepository;
+  private ConnectorBuilderProjectsHandler connectorBuilderProjectsHandler;
+  private Supplier<UUID> uuidSupplier;
+  private UUID workspaceId;
+
+  @SuppressWarnings("unchecked")
+  @BeforeEach
+  void setUp() throws JsonProcessingException {
+    configRepository = mock(ConfigRepository.class);
+    uuidSupplier = mock(Supplier.class);
+    workspaceId = UUID.randomUUID();
+
+    connectorBuilderProjectsHandler = new ConnectorBuilderProjectsHandler(configRepository, uuidSupplier);
+  }
+
+  private ConnectorBuilderProject generateBuilderProject() throws JsonProcessingException {
+    final UUID projectId = UUID.randomUUID();
+    return new ConnectorBuilderProject().withBuilderProjectId(projectId).withWorkspaceId(workspaceId).withName("Test project")
+        .withManifestDraft(new ObjectMapper().readTree("{\"test\": 123}"));
+  }
+
+  @Test
+  @DisplayName("createConnectorBuilderProject should create a new project and return the id")
+  void testCreateConnectorBuilderProject() throws IOException {
+    final ConnectorBuilderProject project = generateBuilderProject();
+
+    when(uuidSupplier.get()).thenReturn(project.getBuilderProjectId());
+
+    final ConnectorBuilderProjectWithWorkspaceId create = new ConnectorBuilderProjectWithWorkspaceId()
+        .builderProject(new ConnectorBuilderProjectDetails().name(project.getName())
+            .draftManifest(new ObjectMapper().convertValue(project.getManifestDraft(), new TypeReference<Map<String, Object>>() {})))
+        .workspaceId(workspaceId);
+
+    final ConnectorBuilderProjectIdWithWorkspaceId response = connectorBuilderProjectsHandler.createConnectorBuilderProject(create);
+    assertEquals(response.getBuilderProjectId(), project.getBuilderProjectId());
+    assertEquals(response.getWorkspaceId(), project.getWorkspaceId());
+
+    verify(configRepository, times(1))
+        .writeBuilderProject(
+            project);
+  }
+
+  @Test
+  @DisplayName("updateConnectorBuilderProject should update an existing project and return the id")
+  void testUpdateConnectorBuilderProject() throws IOException, JsonValidationException, ConfigNotFoundException {
+    final ConnectorBuilderProject project = generateBuilderProject();
+
+    when(configRepository.getConnectorBuilderProject(project.getBuilderProjectId(), false)).thenReturn(project);
+
+    final ExistingConnectorBuilderProjectWithWorkspaceId update = new ExistingConnectorBuilderProjectWithWorkspaceId()
+        .builderProject(new ConnectorBuilderProjectDetails().name(project.getName())
+            .draftManifest(new ObjectMapper().convertValue(project.getManifestDraft(), new TypeReference<Map<String, Object>>() {})))
+        .workspaceId(workspaceId).builderProjectId(project.getBuilderProjectId());
+
+    connectorBuilderProjectsHandler.updateConnectorBuilderProject(update);
+
+    verify(uuidSupplier, never()).get();
```

Comment:
Why is this important for this test? If feels like it's not a relevant outcome if it's called or not

Reply:
Fair point, I'm going to remove this.